SharpPcap/LibPcap/LibPcapSafeNativeMethods.Encoding.cs

```diff
@@ -48,6 +48,66 @@ private static Encoding ConfigureStringEncoding()
             return Encoding.GetEncoding(0);
         }
 
+        /// <summary>
+        /// Helper class to marshall StringBuilder depending on encoding used by Libpcap
+        /// </summary>
+        private class PcapStringBuilderMarshaler : ICustomMarshaler
```

Comment:
I would do special marshaller for error buffer, than using `StringBuilder`. `StringBuilder` creates always an array of 16 chars, see https://github.com/dotnet/runtime/blob/cdc8418a7f4e51b771db2ae7ee5cde5f479cde7e/src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs#L79.

like a custom object instead of `StringBuilder`:
```
class ErrorBuffer
{
    public string buf = "";
}
```
in `MarshalNativeToManaged` you can fill then `buf`. This improvement here is you have already a string object in hand, which was already created in `MarshalNativeToManaged`.

Reply:
The ICustomMarshaler was giving me too much headache, and I figured a way to do a `struct ErrorBuffer` without having to deal with ICustomMarshaler

Let me know if it works for you.

Reply:
I added my review suggestion at https://github.com/trivalik/sharppcap/commit/22e306165bab0c1a6fa02fd032960373f8f448fa. Basically the added gcHandle line is the only real problem.

Reply:
Good catch, fixed